src/main/java/spike/parser/Parser.java

```diff
@@ -0,0 +1,238 @@
+        }
+        return new FindCommand(inputLine.substring(5));
+    }
+
+
+    private Command parseList(String inputLine) throws SpikeException {
+        if (inputLine.length() >= 5) {
+            // User tries to list task by date
+            DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+            LocalDateTime ldt = parseDateTime(inputLine.substring(5), dtf);
+            if (ldt == null) {
+                throw new SpikeException("Kindly enter the date in the format yyyy-MM-dd 0000 to filter by date");
+            }
+            return new ListCommand(0, ldt);
+        } else {
+            return new ListCommand();
+        }
+    }
+
+    private Command parseAdd(CommandName c, String command, String[] commandWords) throws SpikeException {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        switch (c) {
+        case TODO:
+            if (command.length() <= 5) {
+                throw new SpikeException("Hmmmm what to do? Think again?");
+            }
+            ToDo newTD = new ToDo(command.substring(command.indexOf("todo") + 5));
+            return new AddCommand(newTD);
+        case DEADLINE:
+            // Extract description and deadline and pass to constructor
+            if (commandWords.length <= 2 || command.indexOf("/by") == -1
+                    || commandWords[1].equals("/by") || command.indexOf("/by") + 3 == command.length()) {
+                throw new SpikeException("Deadline or task description missing.");
+            }
+            LocalDateTime deadlineT = parseDateTime(command.substring(command.indexOf("/by") + 4), dtf);
+            if (!(deadlineT == null)) {
+                Deadline newD = new Deadline(command.substring(command.indexOf("deadline") + 9,
+                        command.indexOf("/by") - 1), deadlineT);
+                return new AddCommand(newD);
+            } else {
+                throw new SpikeException("Please enter a valid date in the format yyyy-MM-dd HHmm");
+            }
+        case EVENT:
+            if (commandWords.length <= 2 || command.indexOf("/at") == -1
+                    || commandWords[1].equals("/at") || command.indexOf("/at") + 3 == command.length()) {
+                throw new SpikeException("Event time or event description missing.");
+            }
+            LocalDateTime eventT = parseDateTime(command.substring(command.indexOf("/at") + 4), dtf);
+            if (!(eventT == null)) {
+                Event newE = new Event(command.substring(command.indexOf("event") + 6,
+                        command.indexOf("/at") - 1), eventT);
+                return new AddCommand(newE);
+            } else {
+                throw new SpikeException("Please enter a valid date in the format yyyy-MM-dd HHmm");
+            }
+        default:
+            return new IncorrectCommand("Should not reach this line");
+        }
+    }
```

Comment:
- Missing Javadocs comments
- You might want to put some line breaks to improve readability